text/0041-fixed-point.md

```diff
@@ -0,0 +1,104 @@
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [amaranth-lang/rfcs#41](https://github.com/amaranth-lang/rfcs/pull/41)
+- Amaranth Issue: [amaranth-lang/amaranth#0000](https://github.com/amaranth-lang/amaranth/issues/0000)
+
+# Fixed point types
+
+## Summary
+[summary]: #summary
+
+Add fixed point types to Amaranth.
+
+## Motivation
+[motivation]: #motivation
+
+Fractional values in hardware are usually represented as some form of fixed point value.
+Without a first class fixed point type, the user has to manually manage how the value needs to be shifted to be represented as an integer and keep track of how that interacts with arithmetic operations.
+
+A fixed point type would encode and keep track of the precision through arithmetic operations, as well as provide standard operations for converting values to and from fixed point representations with correct rounding.
+
+## Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+TODO
+
+## Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+This RFC proposes a library addition `amaranth.lib.fixedpoint` with the following contents:
+
+`FixedPoint` is a `ShapeCastable` subclass.
+The following operations are defined on it:
+
+- `FixedPoint(f_width, /, *, signed)`: Create a `FixedPoint` with `f_width` fractional bits.
+- `FixedPoint(i_width, f_width, /, *, signed)`: Create a `FixedPoint` with `i_width` integer bits and `f_width` fractional bits.
+- `FixedPoint.cast(shape)`: Cast `shape` to a `FixedPoint` instance.
+- `.i_width`, `.f_width`, `.signed`: Width and signedness properties.
+- `.const(value)`: Create a fixed point constant from an `int` or `float`, rounded to the closest representable value.
+- `.as_shape()`: Return the underlying `Shape`.
+- `.__call__(target)`: Create a `FixedPointValue` over `target`.
+
+`Q(*args)` is an alias for `FixedPoint(*args, signed=True)`.
```

Comment:
I understand Q-notation is an established one, however since Amaranth defaults to unsigned, I feel that having `SQ` and `UQ` (without having just `Q`) would serve Amaranth better.